templates/types/streaming/nextjs/app/components/ui/chat/chat-message/chat-agent-events.tsx

```diff
@@ -0,0 +1,60 @@
+import { icons } from "lucide-react";
+import { useMemo } from "react";
+import { AgentEventData } from "../index";
+
+type MergedEvent = {
+  agent: string;
+  texts: string[];
+  iconName?: string; // lucide icon name
```

Comment:
@thucpn i am writing now the backend code - on a second thought, I think it's better we add the icon lookup (by agent name)  here - then people can easily change the icon library